### PR TITLE
consul: 1.10.3 -> 1.11.1

### PR DESCRIPTION
**Pull request by techknowlogick**
###### Motivation for this change
Updated consul to latest version which supports darwin/arm64 (hopefully this can be backported, although I'm new to nix and am unsure of exact policies on this)

###### Things done
Tests run & verified binary runs on darwin/arm64

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -11617,6 +11617,12 @@
     githubId = 2389333;
     name = "Andy Tockman";
   };
+  techknowlogick = {
+    email = "techknowlogick@gitea.io";
+    github = "techknowlogick";
+    githubId = 164197;
+    name = "techknowlogick";
+  };
```
Comment (nh2) [inline maintainers/maintainer-list.nix:11625]: Please split this into a separate commit in the same PR, so that if the consul commit has to be reverted for some reason, you stay in this list.

File: pkgs/servers/consul/default.nix [hunk 2]
```diff
@@ -26,11 +26,11 @@ buildGoModule rec {
   # has a split module structure in one repo
   subPackages = ["." "connect/certgen"];
 
-  vendorSha256 = "sha256-cQP1po9LGunFVocl4+HPs67oae2KpgyfRRB+xGVySUY=";
+  vendorSha256 = "09rz2xv407ym71dap7f6bbqhdnqvylvbd9zg6f6h7qsb88nvyzsp";
 
   doCheck = false;
 
-  deleteVendor = true;
+  deleteVendor = false;
```
Comment (nh2) [inline pkgs/servers/consul/default.nix:33]: Can you add a comment why this is needed?

**Review by nh2 — COMMENTED** [review]
> latest version which supports darwin/arm64 (hopefully this can be backported, although I'm new to nix and am unsure of exact policies on this)

In general we don't backport major updates, and Consul in particular is problematic, because it gives no forward compat at all, meaning that if you upgrade from 1.10 to 1.11, and something breaks, and you decide to roll back, the on-disk data was already permanently upgraded and the old version will refuse to read it.

In any case, where did you get the info about

> supports darwin/arm64

from? Maybe a newer 1.10 point release also provides that? That one we could backport to 21.11.

**Comment by techknowlogick** [comment]
@nh2 thank you for your feedback :)

> In any case, where did you get the info about

from https://github.com/hashicorp/consul/issues/9852, even though looking at the related PR it only builds the binary. I am unable to get it to build from 1.10.3 (I will test latest tag in 1.10.x branch to see if that works on darwin/arm64, to see if there is something that could be backported).

> Please split this into a separate commit in the same PR, so that if the consul commit has to be reverted for some reason, you stay in this list.

Done.

**Review by nh2 — APPROVED** [review]

**Comment by nh2** [comment]
Thanks for your contribution!
